drivers/adc/adc_demo/adc_demo.c

```diff
@@ -85,6 +110,81 @@ int32_t adc_demo_remove(struct adc_demo_desc *desc)
 	return SUCCESS;
 }
 
+int32_t adc_get_number_of_channels(void* dev)
+{
+	struct adc_demo_desc *desc = dev;
```

Comment:
check for 
```
if (!dev)
       return -ENODEV;
```
Do this for all other functions

Reply:
Not sure where this function will be called do you check there that dev is not NULL ?  If you do this before calling this function then you don't need it here.

Reply:
updated